lib/chef/resource/windows_workgroup.rb

```diff
@@ -0,0 +1,100 @@
+#
+# Author:: Derek Groh (<derekgroh@github.io>)
+# Copyright:: 2018, Derek Groh
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+require "chef/mixin/powershell_out"
+
+class Chef
+  class Resource
+    class WindowsWorkgroup < Chef::Resource
+      resource_name :windows_workgroup
+      provides :windows_workgroup
+
+      include Chef::Mixin::PowershellOut
+
+      description "Use the windows_workgroup resource to join change the workgroup of a machine."
+      introduced "14.5"
+
+      property :workgroup_name, String,
+               description: "The name of the workgroup for the computer.",
+               validation_message: "The 'workgroup_name' property must not contain spaces.",
+               regex: /^\S*$/, # no spaces
+               name_property: true
+
+      property :user, String,
+               description: "The local administrator user to use to change the workgroup."
+
+      property :password, String,
+               description: "The password for the local administrator user."
+
+      property :reboot, Symbol,
+               equal_to: [:immediate, :delayed, :never, :request_reboot, :reboot_now],
+               validation_message: "The reboot property accepts :immediate (reboot as soon as the resource completes), :delayed (reboot once the Chef run completes), and :never (Don't reboot)",
+               description: "Controls the system reboot behavior post workgroup joining. Reboot immediately, after the Chef run completes, or never. Note that a reboot is necessary for changes to take effect.",
+               default: :immediate
+
+      # define this again so we can default it to true. Otherwise failures print the password
+      property :sensitive, [TrueClass, FalseClass],
+               default: true
+
+      action :join do
+        description "Update the workgroup."
+
+        unless workgroup_member?
+          cmd = "$pswd = ConvertTo-SecureString \'#{new_resource.password}\' -AsPlainText -Force;" if new_resource.password
```

Comment:
As new_resource.password is conditional, `cmd` should be instantiated separately or the following lines will fail.

Reply:
i.e. 
```
cmd = ""
cmd << "whatever" if whatever
```